python/adjoint/optimization_problem.py

```diff
@@ -123,7 +166,29 @@ def __call__(
         need_gradient: bool = True,
         beta: float = None,
     ) -> Tuple[List[np.ndarray], List[List[np.ndarray]]]:
-        """Evaluate value and/or gradient of objective function."""
+        """Evaluate value and/or gradient of objective function.
+
+        Args:
+            rho_vector: lists of design variables. Each list represents design variables
+              of one design region. The design is updated to the specified values; functions
+              and gradients will then be evaluated at this configuration of design variables.
+            need_value: whether forward evaluations are needed. Default is True.
+            need_gradient: whether adjoint and gradients evaluatiosn are needed. Default is True.
+            beta: the strength of projection of rho_vector. Default to None.
+
+        Returns:
+            A tuple (f0, gradient) where:
+            f0 is the list of objective functions values when design variables
+              are set to rho_vector
+            gradient is the lists of gradients of objective functions with respect to
```

Comment:
Clarify that `gradient` is a list (over objective functions) of lists (over design regions) of 2d arrays (design vars x frequencies) of derivatives.   If there is only a single objective function, the outer 1-element list is replaced by just that element, and similarly if there is only one design region, while if there is only 1 frequency then the innermost array is squeezed to a 1d array.  (So, for example, if you have only a single objective function, a single design region, and a single frequency, then `gradient` is simply a 1d array of the derivatives.)